### ISSUE TITLE
crawling process halt when scheduled task for link crawler cron changed to sometime later

### ISSUE THREAD

**Issue by allison-soo** [central-queensland-uni/moodle-tool_crawler#25]
I realised that once I changed the link crawler robot (\tool_crawler\task\crawl_task) cron to run every Sat instead ASAP under Server->Scheduled Tasks, the currently crawling process will halt.

I am expecting the currently crawling cycle will complete and follow new scheduled tasks settings in next cycle.  A bug?

**Comment by brendanheywood** [comment]
No this is how it is supposed to work. The crawling is broken up into many tiny processes to make it more robust and spread the load.

What are you trying to achieve? Do you want it to crawl to completion and then stop, and then only kick off a new crawl session once a week?

**Comment by allison-soo** [comment]
Yes. New crawl session once a week in the weekend will be sufficient at this stage.

I have few completed cycle now and would like to involve the development team to 'correct' any broken links found.

**Comment by brendanheywood** [comment]
Ok so only starting a crawl at a set time would be a new feature. But a half workaround could be to run the crawler every 5 minutes on sat and sunday. If  you know it typically completes in a day or two this would give you pretty well what you want, ie a full run every week, but only running during weekends. I will rename this issue and turn it into a new feature request
